pkg/service/auth_context.go

```diff
@@ -63,53 +62,119 @@ func (authContext *AuthContext) evaluateAuthConfig(ctx context.Context, config c
 	}
 }
 
-func (authContext *AuthContext) evaluateAuthConfigs(authConfigs []common.AuthConfigEvaluator, cb configCallback) error {
-	errGroup, ctx := errgroup.WithContext(context.Background())
+func (authContext *AuthContext) evaluateOneAuthConfig(authConfigs []common.AuthConfigEvaluator, cb evaluateCallback) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	waitGroup := new(sync.WaitGroup)
+	size := len(authConfigs)
+	errorChannel := make(chan error, size)
+	successChannel := make(chan bool, size)
+
+	waitGroup.Add(size)
+
+	go func() {
+		waitGroup.Wait()
+		close(errorChannel)
+		close(successChannel)
+	}()
 
 	for _, authConfig := range authConfigs {
 		objConfig := authConfig
-		errGroup.Go(func() error {
-			return authContext.evaluateAuthConfig(ctx, objConfig, cb)
-		})
+		go func() {
+			defer waitGroup.Done()
+
+			err := authContext.evaluateAuthConfig(ctx, objConfig, cb)
+			if err != nil {
+				errorChannel <- err
+			} else {
+				successChannel <- true
+				cancel() // cancels the context if at least one thread succeeds
```

Comment:
this marks the context as cancelled. Are other jobs checking if the context is cancelled?

Reply:
Yes. Whenever there are more than one config of the same auth step (i.e. more than one identify config, more than one metadata config, more than one authorization config), those will run concurrently to each other, and line 51 of this same file incurs of signalling to other when the context has been cancelled.